### PR TITLE
Add dataframe.

### PR DESCRIPTION
**Pull request by LaurentMazare**
The dataframe package has a small implementation of a dataframe similar to pandas in pure ocaml and providing more type checking guarantees.

### CODE REVIEW

File: packages/dataframe/dataframe.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
+opam-version: "2.0"
+name:         "dataframe"
+bug-reports:  "https://github.com/LaurentMazare/ocaml-dataframe/issues"
+homepage:     "https://github.com/LaurentMazare/ocaml-dataframe"
+dev-repo:     "git+https://github.com/LaurentMazare/ocaml-dataframe.git"
+maintainer:   "Laurent Mazare <lmazare@gmail.com>"
+authors:      [ "Laurent Mazare" ]
+
+build: [["dune" "build" "-j" jobs "@install"]]
```
Comment (mseri) [inline packages/dataframe/dataframe.0.1/opam]:
```suggestion
build: [["dune" "build" "-j" jobs "-p" name]]
```

File: packages/dataframe/dataframe.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
+opam-version: "2.0"
+name:         "dataframe"
```
Comment (mseri) [inline packages/dataframe/dataframe.0.1/opam]: Name and version are redundant, opam will override them with the name and version from the path

```suggestion
```

**Comment by mseri** [comment]
Currently fails with 

```
#=== ERROR while compiling dataframe.0.1 ======================================#
# context              2.0.4 | linux/x86_64 | ocaml-base-compiler.4.08.0 | pinned(https://github.com/LaurentMazare/ocaml-dataframe/archive/0.1.tar.gz)
# path                 ~/.opam/4.08/.opam-switch/build/dataframe.0.1
# command              ~/.opam/4.08/bin/dune build -j 71 @install
# exit-code            1
# env-file             ~/.opam/log/dataframe-15-58acc7.env
# output-file          ~/.opam/log/dataframe-15-58acc7.out
### output ###
# File "src/dataframe/df.ml", line 338, characters 16-28:
# 338 |         let v = R.float name in
#                       ^^^^^^^^^^^^
# Error (warning 66): unused open! R.Let_syntax.Let_syntax.Open_on_rhs.
# File "src/dataframe/df.ml", line 356, characters 16-26:
# 356 |         let v = R.int name in
#                       ^^^^^^^^^^
# Error (warning 66): unused open! R.Let_syntax.Let_syntax.Open_on_rhs.
```

**Comment by kit-ty-kate** [comment]
The `v` prefix in janestreet packages are relevant. Good otherwise. Thanks!

**Pull request merged**